components/script/dom/window.rs

```diff
@@ -1980,6 +1980,15 @@ impl Window {
             );
         }
 
+        let document = self.Document();
+        let pending_web_fonts = self.layout.borrow().waiting_for_web_fonts_to_load();
+        let font_face_set = document.Fonts();
+        let is_ready_state_complete = document.ReadyState() == DocumentReadyState::Complete;
+
+        if !pending_web_fonts && is_ready_state_complete {
+            font_face_set.fulfill_ready_promise_if_needed();
```

Comment:
I don't think this is correct, instead the promise must be fulfilled as part of the [`switch-the-fontfaceset-to-loaded`](https://drafts.csswg.org/css-font-loading/#switch-the-fontfaceset-to-loaded). However the queuing of some of the task that call into this mechanism seems to originate in methods on `FontFace`(the constructor [for example](https://drafts.csswg.org/css-font-loading/#ref-for-switch-the-fontfaceset-to-loaded)), so perhaps some kind of partial implementation of that interface should be part of this PR.

Reply:
I think @mrobinson and I discussed this approach during implementation but decided not to have a new task because currently we are only interested in `document.font.ready`, so implementing step 3 of `switch-the-fontfaceset-to-loaded` (Fulfill font face set’s `[[ReadyPromise]]` attribute’s value with font face set) is necessary but not step 4.

I'm not too familiar with the working of our task queue implementation or the specification, so I'll discuss this again with Martin.

Reply:
Ok I see. See my other comment then for a more direct temporary solution.